lib/webmock/request_pattern.rb

```diff
@@ -181,9 +181,8 @@ def to_s
     private
 
     def matches_with_variations?(uri)
-      normalized_template = Addressable::Template.new(WebMock::Util::URI.heuristic_parse(@pattern.pattern))
-
-      WebMock::Util::URI.variations_of_uri_as_strings(uri).any? { |u| normalized_template.match(u) }
+      template = Addressable::Template.new(uri)
```

Comment:
@valscion I believe that with this change you are ignoring `@pattern.pattern` and soimply matching the passed uri with the pattern generated with the same uri, which obviously matches :) feel free to add a spec with webmock pattern that doesn't match your url (e.g different) domain and this code will still match.

Reply:
Perhaps we should silently rescue InvalidURiError and try again with not normalized version of the pattern? 
@rafaelfranca since you are also dealing with similar issue, any thoughts on that?

Reply:
Oh true -- seems like we indeed need some negative assertion specs instead of only positive cases here ☺️. I was quite surprised that none of the existing specs failed with this change here